src/prefect/deployments/base.py

```diff
@@ -424,7 +417,7 @@ def _format_deployment_for_saving_to_prefect_file(
                 schedule_config = _interval_schedule_to_dict(
                     deployment_schedule.schedule
                 )
-            elif isinstance(deployment_schedule.schedule, BaseModel):
+            else:  # all valid schedule types are subclasses of BaseModel
```

Comment:
Hmm, was this here to handle converting the schedules to `dict`s?  Looks like this if-else is about turning schedules of various kinds into `dicts` before serializing them to YAML?

Reply:
yeah that was my understanding too, i guess i just wasn't seeing how we'd ever end up with anything other than a BaseModel subclass for deployment_schedule.schedule since all SCHEDULE_TYPES seem to be BaseModel subclasses

Reply:
Yeah that doesn't really track for me either :thinking: 

So there's also an if-else like this just above [at line 405 in your diff](https://github.com/PrefectHQ/prefect/pull/12639/files#diff-e9a4231fd7a2e33389b0308c100f3b76f58215c3ae881ef9de35f69d1394fd24R405) so we should probably treat these both the same way?

One thing that's concerning is that this function is just typed with dicts:

```
def _format_deployment_for_saving_to_prefect_file(
    deployment: Dict,
) -> Dict:
```

So are we sure these are always parsed into pydantic models before they come through here?  I'll admit I'm not very familiar with this area

Reply:
As the writer of this code, I too am confused about the `isinstance`, `else` should be fine the `MinimalDeploymentSchedule` can't have anything except the schedule types.

Reply:
The only place this is called is from the `deploy` command here: https://github.com/PrefectHQ/prefect/blob/main/src/prefect/cli/deploy.py#L1601

And the deploy command does a lot to ensure that anything getting to that point is a `MinimalDeploymentSchedule`: https://github.com/PrefectHQ/prefect/blob/main/src/prefect/cli/deploy.py#L815-L842

Reply:
cool, that's in line with what I was seeing - good callout on consistent handling. updated in [9cd4aff](https://github.com/PrefectHQ/prefect/pull/12639/commits/9cd4affddcfeac456f3a93c05273feab634c3bb6)